citrination_client/views/client.py

```diff
@@ -12,3 +14,139 @@ def __init__(self, api_key, webserver_host="https://citrination.com", suppress_w
 
     def create(self, ml_template, search_template):
         pass
+
+    def get_available_columns(self, dataset_ids):
+        """
+        Retrieves the set of columns from the combination of dataset ids given
+
+        :param dataset_ids: The id of the dataset to retrieve columns from
+        :type dataset_ids: list of int
+        :return: A list of column names from the dataset ids given.
+        :rtype: list of str
+        """
+
+        data = {
+            "dataset_ids":
+                dataset_ids
+        }
+
+        failure_message = "Failed to get available columns in dataset(s) {}".format(dataset_ids)
+
+        return self._get_success_json(self._post_json(
+            'datasets/get-available-columns', data, failure_message=failure_message))['data']
+
+    def generate_search_template(self, dataset_ids):
```

Comment:
This isn't supposed to be user-facing, is it? If not, let's hide it by prepending `__` (aka `__generate_base_search_template`)